src/components/auth/SignUp.js

```diff
@@ -0,0 +1,172 @@
+import React, { useEffect, useState } from 'react';
+import { Link, useNavigate } from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
+import Container from 'react-bootstrap/Container';
+import Form from 'react-bootstrap/Form';
+import Button from 'react-bootstrap/Button';
+import Spinner from 'react-bootstrap/Spinner';
+import { GrFormClose } from 'react-icons/gr';
+import { resetState, signUp } from '../../redux/auth/auth';
+import defaultAvatar from '../../assets/images/profile-pic.png';
+
+const SignUp = () => {
+  const formInitialState = {
+    name: '',
+    email: '',
+    phone: '',
+    password: '',
+    password_confirmation: '',
+    avatar: {
+      value: '',
+      preview: defaultAvatar,
+      file: null,
+    },
+  };
+  const authState = useSelector((state) => state.auth);
+  const [formState, setFormState] = useState(formInitialState);
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+
+  useEffect(() => () => dispatch(resetState()), []);
+
+  useEffect(() => {
+    if (authState.status === 'failed' && (formState.password || formState.password_confirmation)) {
+      setFormState((state) => ({
+        ...state,
+        password: '',
+        password_confirmation: '',
+      }));
+    }
+    if (authState.userSignedIn) {
+      if (authState.status === 'success') {
+        navigate('/', { state: { notice: 'Account created successfully' } });
+      } else {
+        navigate('/', { state: { notice: 'You are already signed in!' } });
+      }
+    }
+  }, [authState.status]);
+
+  const removeSelectedPicture = () => {
+    setFormState((state) => ({
+      ...state,
+      avatar: formInitialState.avatar,
+    }));
+  };
+
+  const previewHandler = (e) => {
+    if (e.target.value) {
+      const file = e.target.files[0];
+      setFormState((state) => ({
+        ...state,
+        avatar: {
+          value: e.target.value,
+          preview: URL.createObjectURL(file),
+          file,
+        },
+      }));
+      return undefined;
+    }
```

Comment:
3. (a) Moved the extraction of the file outside of the conditional so that the function doesn't need to do more than one 
         return.
     (b) Added a guard clause to exit early if there's no file selected.
     (c) Created a variable imageUrl to store the created object URL, making it easier to read and avoiding duplication.
     (d) Changed the way setFormState is called by using the functional form of setState to avoid potential race 
          conditions. Also, by using the spread operator, we ensure that we're creating a new object rather than mutating 
          the existing one.
     (e) Added prevState.avatar to the spread operator to avoid losing any other properties that might be present in the 
          avatar object.